src/v2/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -146,211 +141,180 @@ export class ShippingRoute extends Component<ShippingProps, ShippingState> {
     }
   }
 
-  handleAddressDelete = (deletedAddressID: string) => {
-    const addressList = this.getAddressList()
+  const handleAddressDelete = (deletedAddressID: string) => {
+    const addressList = getAddressList()
 
     if (!addressList || addressList.length === 0) {
-      this.setState({
-        selectedAddressID: NEW_ADDRESS,
-        shippingQuotes: null,
-        shippingQuoteId: undefined,
-      })
-    } else if (this.state.selectedAddressID == deletedAddressID) {
-      this.selectSavedAddress(
+      setSelectedAddressID(NEW_ADDRESS)
+      setShippingQuotes(null)
+      setShippingQuoteId(undefined)
+    } else if (selectedAddressID == deletedAddressID) {
+      selectSavedAddress(
         addressList.find(address => address?.node?.isDefault)?.node?.internalID!
       )
     }
   }
 
-  componentDidMount() {
-    if (
-      this.isArtaShipping() &&
-      !this.isCreateNewAddress() &&
-      !this.state.shippingQuoteId
-    ) {
-      this.selectShipping()
-    }
-  }
-
-  getAddressList = () => this.props.me.addressConnection?.edges
+  const getAddressList = () => props.me.addressConnection?.edges
 
-  getOrderArtwork = () => this.props.order.lineItems?.edges?.[0]?.node?.artwork
+  const getOrderArtwork = () => props.order.lineItems?.edges?.[0]?.node?.artwork
 
-  isCreateNewAddress = () => this.state.selectedAddressID === NEW_ADDRESS
+  const isCreateNewAddress = () => selectedAddressID === NEW_ADDRESS
 
-  isArtaShipping = () => {
-    const addresses = this.getAddressList()
-    const artaShippingEnabled = !!this.getOrderArtwork()?.artaShippingEnabled
+  const isArtaShipping = () => {
+    const addresses = getAddressList()
+    const artaShippingEnabled = !!getOrderArtwork()?.artaShippingEnabled
 
-    const shippingCountry = this.isCreateNewAddress()
-      ? this.state.address.country
+    const shippingCountry = isCreateNewAddress()
+      ? address.country
       : addresses &&
         addresses.find(
-          address => address?.node?.internalID == this.state.selectedAddressID
+          address => address?.node?.internalID == selectedAddressID
         )?.node?.country
 
     return (
-      this.state.shippingOption === "SHIP" &&
+      shippingOption === "SHIP" &&
       artaShippingEnabled &&
       shippingCountry === "US"
     )
   }
 
-  onContinueButtonPressed = async () => {
-    if (this.isArtaShipping() && !!this.state.shippingQuoteId) {
-      this.selectShippingQuote()
+  useEffect(() => {
+    if (
+      isArtaShipping() &&
+      selectedAddressID &&
+      !shippingQuotes &&
+      !shippingQuoteId
+    ) {
+      selectShipping()
+    }
+  }, [shippingOption, shippingQuotes, shippingQuoteId])
```

Comment:
```suggestion
  }, [])
```

I'm not totally sure about when this effect should run, but in the old version it ran on `componentDidMount`. To replicate that behavior, we should pass an empty array to this dependency array, which would prompt the effect to only run once, when the component mounts.

Reply:
Actually I think I was wrong -- I just found https://github.com/artsy/force/pull/8918/files#diff-3450e2d03077be9b8efde9cbfc7e452910188d0d80ad0cd4848b242d63840b60L484 and now I see why we were firing the effect when any of those change.

Side note....this is a perfect example of how `useEffect` can help keep all these related things in one place instead of two

Reply:
Exactly, this is grouping not only `componentDidMount` but all the callbacks defined on `this.setState()` of the class component.